optuna/integration/pytorch_distributed.py

```diff
@@ -24,6 +25,42 @@
 )
 
 
+def updates_state(f: Callable[..., Any]) -> Callable[..., Any]:
+    """Method decorator to fetch updated trial state from rank 0 after f is run.
+
+    This decorator ensures trial properties (params, distributions, etc.) on all distributed
+    processes are up-to-date with the wrapped trial stored on rank 0.
+    It should be applied to all Trial methods that update property values.
+    """
+
+    @functools.wraps(f)
+    def wrapped(self: Any, *args, **kwargs) -> Any:
+        def state() -> Sequence:
+            assert self._delegate is not None
+            return (
+                self._delegate.number,
+                self._delegate.params,
+                self._delegate.distributions,
+                self._delegate.user_attrs,
+                self._delegate.system_attrs,
+                self._delegate.datetime_start,
+            )
+
+        try:
+            return f(self, *args, **kwargs)
```

Comment:
Mypy showed an error due to missing type annotation for the arguments.
Not sure, but the optuna code corresponding to this `wrapped` function takes `args` and `kwargs`, so I guess we can refactor this function as follows:

```suggestion
    def wrapped(*args: Any, **kwargs: Any) -> Any:
        def state() -> Sequence:
            self: TorchDistributedTrial = args[0]
            assert self._delegate is not None
            return (
                self._delegate.number,
                self._delegate.params,
                self._delegate.distributions,
                self._delegate.user_attrs,
                self._delegate.system_attrs,
                self._delegate.datetime_start,
            )

        try:
            return f(*args, **kwargs)
```

In addition, I guess we can use `properties` instead of `state`. What do you think of it?